### PR TITLE
Issue #105 - PMDise Catalog Metadata

### PR DESCRIPTION
**Pull request by robons**
Added a new CLI command which strips out the DCAT catalog metadata out of the `file.csv-metadata.json` file and drops it into a `file.csv-metadata.json.nq` N-Quads file.

```
$ pmdutils dcat pmdify --help
Usage: pmdutils dcat pmdify [OPTIONS] CSVW_METADATA_FILE BASE_URI
                            DATA_GRAPH_URI CATALOG_METADATA_GRAPH_URI

  Given a local CSV-W metadata.json file, re-work the DCATv2 metadata such
  that it ends up in the format required for PMD.

  Removes the expected DCAT metadata from the CSV-W metadata.json file and
  outputs a new N-Quads file containing PMDCAT metadata at
  <CSVW_METADATA_FILE>.nq
``` 

* `CSVW_METADATA_FILE` - the `.csv-metadata.json` CSV-W file.
* `BASE_URI` - the base URI which URIs inside the CSV-W are relative to (generally the URL of the directory holding the CSV-W).
* `DATA_GRAPH_URI` - the graph URI which the `skos:ConceptScheme`/`qb:DataSet` in to be held within.
* `CATALOG_METADATA_GRAPH_URI` - the graph URI which the DCAT metadata will be uploaded to.

Satisfies Issue #105

### CODE REVIEW

**Comment by canwaf** [comment]
I stripping out the catalog data from a unified file necessary for PMD to process it? @robons

**Comment by robons** [comment]
> I stripping out the catalog data from a unified file necessary for PMD to process it? @robons

Since we're separating it out into a separate graph/file we don't want to duplicate the information we're uploading.

File: csvcubed-pmd/tests/unit/dcatcli/test_pmdify.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    )
+
+    existing_dcat_dataset = pmdify._get_catalog_entry_from_dcat_dataset(csvw_graph)
+
+    assert existing_dcat_dataset is not None
+    assert existing_dcat_dataset.title == "single-measure-bottles-bulletin"
+    assert existing_dcat_dataset.label == "single-measure-bottles-bulletin"
+    assert existing_dcat_dataset.issued == datetime.datetime(2019, 2, 28)
+    assert existing_dcat_dataset.modified == datetime.datetime(2019, 2, 28)
+    assert existing_dcat_dataset.comment == "some comment goes here"
```
Comment (canwaf) [inline csvcubed-pmd/tests/unit/dcatcli/test_pmdify.py:31]: Unit test fails here.

**Review by canwaf — CHANGES_REQUESTED** [review]
The code looks fine, and I can assume it does what it says on the tin (behave says so); however you need to address the following:
- [ ] pyrite fails (6 errors)
- [ ] unit test: test_extracting_metadata(): fails (assertion on comment)

The cli structure is getting unwieldly where there are keywords after keywords after keywords. That said the helpfile should mention that input isn't exclusively metadata, and that the metadata gets removed from the source and output is multiple files.

File: csvcubed-pmd/tests/unit/dcatcli/test_pmdify.py (hunk 1 above)
Comment (robons) [reply csvcubed-pmd/tests/unit/dcatcli/test_pmdify.py:31]: fixed

**Review by canwaf — APPROVED** [review]
Thanks for fixing it all.

**Pull request merged**